src/main/java/pro/sky/telegrambot/listener/TelegramBotUpdatesListener.java

```diff
@@ -3,34 +3,107 @@
 import com.pengrad.telegrambot.TelegramBot;
 import com.pengrad.telegrambot.UpdatesListener;
 import com.pengrad.telegrambot.model.Update;
+import com.pengrad.telegrambot.request.SendMessage;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
+import pro.sky.telegrambot.model.NotificationTask;
+import pro.sky.telegrambot.repository.RepositoryNotificationTask;
 
 import javax.annotation.PostConstruct;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
 import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 @Service
 public class TelegramBotUpdatesListener implements UpdatesListener {
 
-    private Logger logger = LoggerFactory.getLogger(TelegramBotUpdatesListener.class);
+    private static final Logger logger = LoggerFactory.getLogger(TelegramBotUpdatesListener.class);
+    //создаем статический паттерн с регулярным выражением
+    private static final Pattern MESSAGE_PATTERN = Pattern.compile("([0-9\\.:\\s]{16})(\\s)(.+)");
+    //создаем статическое поле с паттерном для даты и времени
+    private static final DateTimeFormatter DATE_TIME_FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy HH:mm");
+    private final RepositoryNotificationTask repositoryNotificationTask;
 
     @Autowired
     private TelegramBot telegramBot;
 
+    public TelegramBotUpdatesListener(RepositoryNotificationTask repositoryNotificationTask) {
+        this.repositoryNotificationTask = repositoryNotificationTask;
+    }
+
     @PostConstruct
     public void init() {
         telegramBot.setUpdatesListener(this);
     }
 
+    //этот метод реализует интерфейс и вызывается каждый раз когда приходит обновление и после вызывает метод processUpdate
     @Override
     public int process(List<Update> updates) {
-        updates.forEach(update -> {
-            logger.info("Processing update: {}", update);
-            // Process your updates here
-        });
+        updates.forEach(this::processUpdate);
         return UpdatesListener.CONFIRMED_UPDATES_ALL;
     }
 
+    //этот метод проверят, содержится ли сообщение с форматом, что нужен для записи напоминания или команда /start
+    private void processUpdate(Update update) {
+        logger.info("Processing update: {}", update);
+        if (update.message() == null) return;
```

Comment:
```suggestion
        if (update.message() == null) {
            return;
        }
```
фигурные скобки надо всегда писать